<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainSkel</class>
 <widget class="QMainWindow" name="mainSkel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1013</width>
    <height>672</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>HotSpotter</string>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QTabWidget" name="tablesTabWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>10</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <widget class="QWidget" name="image_view">
        <property name="minimumSize">
         <size>
          <width>445</width>
          <height>0</height>
         </size>
        </property>
        <attribute name="title">
         <string>Image Table</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QTableWidget" name="gxs_TBL">
             <property name="dragEnabled">
              <bool>false</bool>
             </property>
             <property name="sortingEnabled">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="chip_view">
        <attribute name="title">
         <string>Chip Table</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QTableWidget" name="cxs_TBL">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="sortingEnabled">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="name_view">
        <attribute name="title">
         <string>Name Table</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QTableWidget" name="nxs_TBL"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="result_view">
        <attribute name="title">
         <string>Query Results Table</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QTableWidget" name="res_TBL">
             <property name="selectionMode">
              <enum>QAbstractItemView::ExtendedSelection</enum>
             </property>
             <property name="textElideMode">
              <enum>Qt::ElideLeft</enum>
             </property>
             <property name="sortingEnabled">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QTextEdit" name="outputEdit">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>1</verstretch>
        </sizepolicy>
       </property>
       <property name="acceptRichText">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QProgressBar" name="progressBar">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>1</verstretch>
        </sizepolicy>
       </property>
       <property name="value">
        <number>24</number>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1013</width>
     <height>23</height>
    </rect>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::DefaultContextMenu</enum>
   </property>
   <property name="defaultUp">
    <bool>false</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>false</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew_Database"/>
    <addaction name="actionOpen_Database"/>
    <addaction name="separator"/>
    <addaction name="actionSave_Database"/>
    <addaction name="separator"/>
    <addaction name="actionImport_Img_file"/>
    <addaction name="actionImport_Img_dir"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>Options</string>
    </property>
    <addaction name="actionLayout_Figures"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionView_Docs"/>
    <addaction name="separator"/>
    <addaction name="actionView_DBDir"/>
    <addaction name="actionView_Computed_Dir"/>
    <addaction name="actionView_Global_Dir"/>
    <addaction name="separator"/>
    <addaction name="actionWriteLogs"/>
    <addaction name="separator"/>
    <addaction name="actionDelete_Precomputed_Results"/>
    <addaction name="actionDelete_computed_directory"/>
    <addaction name="actionDelete_global_preferences"/>
    <addaction name="separator"/>
    <addaction name="actionDev_Mode_IPython"/>
    <addaction name="actionDeveloper_Reload"/>
   </widget>
   <widget class="QMenu" name="menuActions">
    <property name="title">
     <string>Actions</string>
    </property>
    <addaction name="actionAdd_Chip"/>
    <addaction name="actionNew_Chip_Property"/>
    <addaction name="separator"/>
    <addaction name="actionQuery"/>
    <addaction name="separator"/>
    <addaction name="actionReselect_ROI"/>
    <addaction name="actionReselect_Ori"/>
    <addaction name="separator"/>
    <addaction name="actionNext"/>
    <addaction name="separator"/>
    <addaction name="actionDelete_Chip"/>
    <addaction name="actionDelete_Image"/>
   </widget>
   <widget class="QMenu" name="menuBatch">
    <property name="title">
     <string>Batch</string>
    </property>
    <addaction name="actionPrecomputeChipsFeatures"/>
    <addaction name="actionPrecompute_Queries"/>
    <addaction name="separator"/>
    <addaction name="actionScale_all_ROIS"/>
    <addaction name="separator"/>
    <addaction name="actionConvert_all_images_into_chips"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuActions"/>
   <addaction name="menuBatch"/>
   <addaction name="menuOptions"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionOpen_Database">
   <property name="text">
    <string>Open Database</string>
   </property>
   <property name="toolTip">
    <string>Opens a different database directory. </string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionSave_Database">
   <property name="text">
    <string>Save Database</string>
   </property>
   <property name="iconText">
    <string>Save Database</string>
   </property>
   <property name="toolTip">
    <string>Saves the added images / chip properties / and changed names to the database csv tables.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionImport_Img_file">
   <property name="text">
    <string>Import Images (select file(s))</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionOpen_Data_Directory">
   <property name="text">
    <string>View Data Directory</string>
   </property>
  </action>
  <action name="actionOpen_Source_Directory">
   <property name="text">
    <string>View Source Directory</string>
   </property>
  </action>
  <action name="actionUndockDisplay">
   <property name="text">
    <string>Undock Display</string>
   </property>
  </action>
  <action name="actionTogPlt">
   <property name="text">
    <string>Toggle PlotWidget</string>
   </property>
  </action>
  <action name="actionHelpCMD">
   <property name="text">
    <string>Command Line Help</string>
   </property>
  </action>
  <action name="actionHelpGUI">
   <property name="text">
    <string>GUI Help</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionHelpWorkflow">
   <property name="text">
    <string>Workflow Help</string>
   </property>
  </action>
  <action name="actionAdd_Chip">
   <property name="text">
    <string>Add Chip</string>
   </property>
   <property name="toolTip">
    <string>When adding a chip, you select an ROI in Image View. The ROI defines a new chip and it is added (but not saved) to the database.</string>
   </property>
   <property name="shortcut">
    <string>A</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionReselect_ROI">
   <property name="text">
    <string>Reselect ROI</string>
   </property>
   <property name="toolTip">
    <string>Changes the ROI of a chip. Previously associated results and chip data are removed and recomputed. </string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionNext">
   <property name="text">
    <string>Select Next</string>
   </property>
   <property name="toolTip">
    <string>Selects the next image or chip in the database. </string>
   </property>
   <property name="shortcut">
    <string>N</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionDelete_Chip">
   <property name="text">
    <string>Delete Chip</string>
   </property>
   <property name="toolTip">
    <string>Removes a chip from the database. </string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Del</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionQuery">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Query</string>
   </property>
   <property name="toolTip">
    <string>Issues the currently selected chip as a query. The result table is then populated.</string>
   </property>
   <property name="shortcut">
    <string>Q</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionPrev">
   <property name="text">
    <string>Prev</string>
   </property>
  </action>
  <action name="actionWriteLogs">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Write Logs</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionOpen_Internal_Directory">
   <property name="text">
    <string>View Internal Directory</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Edit Preferences</string>
   </property>
   <property name="toolTip">
    <string>Changes algorithm parameters and program behavior.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="toolTip">
    <string>Exits the program</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionConvertImage2Chip">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Convert All Images to Chips</string>
   </property>
  </action>
  <action name="actionBatch_Change_Name">
   <property name="text">
    <string>Batch Change Name</string>
   </property>
  </action>
  <action name="actionReselect_Ori">
   <property name="text">
    <string>Reselect Orientation</string>
   </property>
   <property name="toolTip">
    <string>Changes the orientation of a chip. Previously associated results and chip data are removed and recomputed. </string>
   </property>
   <property name="shortcut">
    <string>O</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionAdd_Metadata_Property">
   <property name="text">
    <string>Add Chip Property</string>
   </property>
  </action>
  <action name="actionAutoassign">
   <property name="text">
    <string>Assign Matches Above Threshold</string>
   </property>
  </action>
  <action name="actionRankErrorExpt">
   <property name="text">
    <string>Rank Error Experiment</string>
   </property>
  </action>
  <action name="actionName_Consistency_Experiment">
   <property name="text">
    <string>Name Consistency</string>
   </property>
  </action>
  <action name="actionIncrease_ROI_Size">
   <property name="text">
    <string>Increase all ROI Sizes</string>
   </property>
  </action>
  <action name="actionView_Docs">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>View Documentation</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionAutoassign_2">
   <property name="text">
    <string>Auto-Assign Matches</string>
   </property>
  </action>
  <action name="actionNew_Database">
   <property name="text">
    <string>New Database</string>
   </property>
   <property name="toolTip">
    <string>Create a new folder to use as a database. </string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionImport_Img_dir">
   <property name="text">
    <string>Import Images (select directory)</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionPrecompute_Queries">
   <property name="text">
    <string>Precompute Queries</string>
   </property>
   <property name="toolTip">
    <string>This might take anywhere from a coffee break to an overnight  procedure depending on how many ROIs you've made. It queries each chip and saves the result which allows multiple queries to be rapidly inspected later. </string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionPrecomputeChipsFeatures">
   <property name="text">
    <string>Precompute Chips/Features</string>
   </property>
   <property name="toolTip">
    <string>Extracts all chips from images, and computes features. This loads the data into memory and reduces waiting time between selecting chips and images.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Return</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionView_DBDir">
   <property name="text">
    <string>View Database Dir</string>
   </property>
   <property name="toolTip">
    <string>Opens the database folder in your operating system's native file browser (explorer/finder/nautilus)</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionScale_all_ROIS">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Scale all ROIs</string>
   </property>
   <property name="toolTip">
    <string>Changes the size of every ROI in the database. </string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionConvert_all_images_into_chips">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Convert all images into chips</string>
   </property>
   <property name="toolTip">
    <string>Every image without a chip has an ROI added to it spanning the entire image.</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionNew_Chip_Property">
   <property name="text">
    <string>New Chip Property</string>
   </property>
   <property name="toolTip">
    <string>Creates a new column in the chip table for user properties.</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionDelete_computed_directory">
   <property name="text">
    <string>Delete Computed Directory</string>
   </property>
   <property name="toolTip">
    <string>Deletes all precomputations (not your ROIs or properties though) and lets all data be recomputed and reloaded. This puts your database in a cleaner state. </string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionDelete_global_preferences">
   <property name="text">
    <string>Delete Global Preferences</string>
   </property>
   <property name="toolTip">
    <string>This deletes global preferences like the previously opened database, your work directory, and defaults algorithm preferences. Use if HotSpotter crashes on loading a database. </string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionView_Computed_Dir">
   <property name="text">
    <string>View Computed Dir</string>
   </property>
   <property name="toolTip">
    <string>Opens the directory where all precomputed chips, query results, and features are stored. </string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionView_Global_Dir">
   <property name="text">
    <string>View Global Dir</string>
   </property>
   <property name="toolTip">
    <string>This is youre ~/.hotspotter folder, which stores non-database-specific preferences.</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionLayout_Figures">
   <property name="text">
    <string>Layout Figures</string>
   </property>
   <property name="toolTip">
    <string>Organizes windows in a grid</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionDev_Mode_IPython">
   <property name="text">
    <string>Developer Mode (IPython)</string>
   </property>
   <property name="toolTip">
    <string>This allows IPython interaction in the terminal and disables GUI interaction until the &quot;exit&quot; command is entered in IPython. </string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+Shift+D</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionDelete_Precomputed_Results">
   <property name="text">
    <string>Delete Cached Query Results</string>
   </property>
   <property name="toolTip">
    <string>Removes all precomputed results. Use if you expect that your results may be corrupted.</string>
   </property>
  </action>
  <action name="actionDeveloper_Reload">
   <property name="text">
    <string>Developer Reload</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+R</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionDelete_Image">
   <property name="text">
    <string>Trash Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Del</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
